app/src/main/java/com/lagradost/cloudstream3/ui/download/DownloadAdapter.kt

```diff
@@ -93,110 +94,128 @@ class DownloadAdapter(
+                downloadHeaderTitle.text = d.name
+                val formattedSizeString = formatShortFileSize(itemView.context, card.totalBytes)
+
+                if (card.child != null) {
+                    downloadHeaderGotoChild.isVisible = false
+
+                    val status = downloadButton.getStatus(card.child.id, card.currentBytes, card.totalBytes)
+                    if (status == DownloadStatusTell.IsDone) {
+                        // We do this here instead if we are finished downloading
+                        // so that we can use the value from the view model
+                        // rather than extra unneeded disk operations and to prevent a
+                        // delay in updating download icon state.
+                        downloadButton.setProgress(card.currentBytes, card.totalBytes)
+                        downloadButton.applyMetaData(card.child.id, card.currentBytes, card.totalBytes)
+                        // We will let the view model handle this
+                        downloadButton.doSetProgress = false
+                        downloadHeaderInfo.text = formattedSizeString
+                    } else downloadButton.doSetProgress = true
+
+                    downloadButton.setDefaultClickListener(card.child, downloadHeaderInfo, mediaClickCallback)
+                    downloadButton.isVisible = true
+
+                    episodeHolder.setOnClickListener {
+                        mediaClickCallback.invoke(DownloadClickEvent(DOWNLOAD_ACTION_PLAY_FILE, card.child))
+                    }
+                } else {
+                    downloadButton.isVisible = false
+                    downloadHeaderGotoChild.isVisible = true
+
+                    try {
+                        downloadHeaderInfo.text = downloadHeaderInfo.context.getString(R.string.extra_info_format)
+                            .format(
+                                card.totalDownloads,
+                                downloadHeaderInfo.context.resources.getQuantityString(
+                                    R.plurals.episodes,
+                                    card.totalDownloads
+                                ),
+                                formattedSizeString
+                            )
+                    } catch (e: Exception) {
+                        // You probably formatted incorrectly
+                        downloadHeaderInfo.text = "Error"
+                        logError(e)
                     }
 
-                    downloadButton.setDefaultClickListener(card.data, downloadChildEpisodeTextExtra, mediaClickCallback)
-
-                    downloadChildEpisodeText.apply {
-                        text = context.getNameFull(d.name, d.episode, d.season)
-                        isSelected = true // Needed for text repeating
+                    episodeHolder.setOnClickListener {
+                        clickCallback.invoke(DownloadHeaderClickEvent(0, d))

```

Comment:
Done

Reply:
@fire-light42 I actually remembered why these didn't use constants, because they didn't do the same thing so it made no sense. So I instead added new constants that are descriptive for what DownloadHeaderClickEvent actually does.